esp_encrypted_img/README.md

```diff
@@ -260,6 +258,42 @@ Make sure these configurations are set correctly in your project's configuration
 
     The public key of the ephemeral server key pair used for ECDH is embedded in the image but not saved to a separate file.
 
+## Per Device Unique Key Support
+
+The `esp_encrypted_img` component supports workflows where each device is provisioned with a unique key pair. This is essential for scenarios requiring per-device image encryption and secure provisioning. By exporting the public key from the device or key management system, you can automate image generation and ensure that only the intended device can decrypt its firmware or data.
+
+It is the application's responsibility to:
+- Generate the key using a good entropy source (the [ESP-IDF Random Number Generation APIs](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/random.html) provide suggested approach using ESP-IDF APIs).
+- Flash the generated key to the device securely (e.g., to eFuse or in Flash).
+- Retrieve the public key (using the API below), send it to the server, and use it to generate pre-encrypted OTA binaries for each device.
+
+<details>
+<summary>How to Export the Public Key</summary>
+
+### Exporting the Public Key Programmatically
+
+The component provides an API to retrieve the public key associated with the decryption context:
+
+```c
+/**
+ * @brief  Export the public key corresponding to the private key.
+ *         The application should free the memory pointed by `pub_key` after use.
+ *         For RSA, the public key is in DER format and corresponds to the private key passed with `esp_encrypted_img_decrypt_start()`.
+ *         For ECIES, the public key is in uncompressed format and is derived from the HMAC key ID passed with `esp_encrypted_img_decrypt_start()`.
```

Comment:
Isn't this standard DER format, would it require some additional processing for the conversion?